fitbenchmarking/utils/tests/test_misc.py

```diff
@@ -53,23 +55,16 @@ def all_neutron_problems(self):
 
         return neutron_problems
 
-
     def all_nist_problems(self):
         """
-        Helper function that returns the names of all NIST problems.
+        Helper function that returns the names of Nist low diff problems.
         """
 
-        nist_problems = [['Misra1a.dat', 'Chwirut2.dat', 'Chwirut1.dat',
-                          'Lanczos3.dat', 'Gauss1.dat', 'Gauss2.dat',
-                          'DanWood.dat', 'Misra1b.dat'],
-                         ['Kirby2.dat', 'Hahn1.dat','MGH17.dat',
-                          'Lanczos1.dat', 'Lanczos2.dat','Gauss3.dat',
-                          'Misra1c.dat', 'Misra1d.dat','ENSO.dat'],
-                         ['MGH09.dat', 'Thurber.dat', 'BoxBOD.dat', 'Rat42.dat',
-                          'MGH10.dat', 'Eckerle4.dat', 'Rat43.dat',
-                          'Bennett5.dat']]
+        nist_ld_problems = [['Misra1a.dat', 'Chwirut2.dat', 'Chwirut1.dat',
```

Comment:
This probably make sense, but just checking the reason for unit testing a subset is that you judge this is sufficient for unit testing, which I likely agree with

Reply:
To make the code a little more general I moved the subfolders out of the NIST data set, thus, I am only checking one of these folders since the data is in the same format

Reply:
OK

Reply:
Although as we get more fitting problem I believe it would make sense to have subfolder with problem sets, hence in that sense I prefer a structure similar to what we had, i.e.
NIST_nonlinear_regression
    average_diff
    high_diff
    low_diff

instead of your new flatter structure

NIST_average_diff
NIST_high_diff
NIST_low_diff

We can discuss this tomorrow in catch up also. As a note I can no longer remember what 'diff' stands for, hence maybe that should be either removed or renamed

Reply:
The above first file structure meant to read
```
NIST_nonlinear_regression
    average_diff
    high_diff
    low_diff
```